backend/controllers/ProductController.js

```diff
@@ -111,7 +118,37 @@ class ProductController {
       });
     }
   }
-  
+
+  static async deleteProduct(req, res) {
+    const productId = req.params.id;
+    const { id } = req.user;
+    const productExists = await ProductService.findProduct(productId);
+    if (productExists === null) {
+      return res.status(404).json({
+        status: 404,
+        message: 'Product doesn\'t exist',
+      });
+    }
+    const isOwner = await ProductService.checkOwner(req.params.id, id);
+    if (!isOwner) {
+      return res.status(401).json({
+        status: 401,
+        message: 'You can not delete this product',
+      });
+    }
+    const deleteStatus = await ProductService.deleteProduct(productId);
+    if (deleteStatus) {
+      return res.status(200).json({
+        status: res.statusCode,
+        message: 'Product deleted successfully',
+        data: deleteStatus,
```

Comment:
I don't think you should return data when something is deleted... I think it only returns a boolean value